quadra/tasks/segmentation.py

```diff
@@ -222,18 +222,22 @@ def __init__(
         self,
         config: DictConfig,
         model_path: str,
-        report_folder: Optional[str] = None,
+        device: str = "cpu",
     ):
-        super().__init__(config=config, model_path=model_path, report_folder=report_folder)
+        super().__init__(config=config, model_path=model_path, device=device)
         self.config = config
-        self.metadata = {"report_files": []}
-
-        # TODO: It's not possible to specify the device from outside!!
-        self.device = utils.get_device(config.trainer.accelerator != "cpu")
 
     def save_config(self) -> None:
         """Skip saving the config."""
 
+    def prepare(self) -> None:
+        """Prepare the evaluation."""
+        super().prepare()
+        self.datamodule = self.config.datamodule
+        # prepare_data() must be explicitly called because there is no training
+        self.datamodule.prepare_data()
```

Comment:
Segmentation datamodule has no check for train or test?

Reply:
I'm having trouble grasping the meaning behind your question. What i see from the other tasks is that whenever you have no lightning training you have to manually call the prepare_data(). I added some comments and TODO, se my commit "todo: Add todo for the prepare_data() manual call"

Reply:
Nada, never review PR at 18:30, I was thinking about setup() :D